saga/inMemorySagaLog.go

```diff
@@ -0,0 +1,72 @@
+package saga
+
+import "errors"
+import "fmt"
+import "sync"
+
+/*
+ * In Memory Implementation of a Saga Log, DOES NOT durably persist messages.
+ * This is for local testing purposes.
+ */
+type inMemorySagaLog struct {
+	sagas map[string]*SagaState
+	mutex *sync.RWMutex
```

Comment:
I don't believe we want locking in this testing/example code. GetSagaState() gives the caller an unlocked pointer to state which LogMessage() can then alter concurrently. The code that operates on Saga structs should lock as necessary instead (assuming we continue to pass by pointer).